custom-events-generated/src/ui.rs

```diff
@@ -0,0 +1,38 @@
+use ratatui::{
+    layout::Alignment,
+    style::{Color, Style},
+    widgets::{Block, BorderType, Paragraph, Widget},
+};
+
+use crate::app::App;
+
+impl Widget for &App {
+    /// Renders the user interface widgets.
+    ///
+    // This is where you add new widgets.
+    // See the following resources:
+    // - https://docs.rs/ratatui/latest/ratatui/widgets/index.html
+    // - https://github.com/ratatui/ratatui/tree/master/examples
+    fn render(self, area: ratatui::prelude::Rect, buf: &mut ratatui::prelude::Buffer)
+    where
+        Self: Sized,
+    {
+        let block = Block::bordered()
+            .title("Template")
+            .title_alignment(Alignment::Center)
+            .border_type(BorderType::Rounded);
+
+        let paragraph = Paragraph::new(format!(
+            "This is a tui template.\n\
+                Press `Esc`, `Ctrl-C` or `q` to stop running.\n\
+                Press left and right to increment and decrement the counter respectively.\n\
+                Counter: {}",
+            self.counter
```

Comment:
Nit: For neatness, I'd probably pull the text out into a string var instead of putting it in the config of the pargraph.

What about using indoc here?

Reply:
Indoc might be a bit overkill. I created a new variable for now.